File: apps/motoboy/app/src/main/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/RouteStopTypeLabels.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

/** Traduz o tipo da parada para o texto exibido na rota do motoboy. */
fun routeStopTypeLabel(type: String): String {
  return ROUTE_STOP_TYPE_LABELS[type] ?: fallbackRouteStopTypeLabel(type)
}

/** Mantem tipos de parada novos legiveis enquanto o app ainda nao conhece o enum. */
private fun fallbackRouteStopTypeLabel(type: String): String {
  val normalized = type.trim().replace("_", " ").lowercase()
  if (normalized.isBlank()) return "Parada"
  return normalized.replaceFirstChar { it.uppercase() }
}

private val ROUTE_STOP_TYPE_LABELS = mapOf(
  "COLETA" to "Coleta",
  "ENTREGA" to "Entrega",
)

File: apps/motoboy-pwa/src/routeStopTypeLabels.ts
const ROUTE_STOP_TYPE_LABELS: Record<string, string> = {
  COLETA: "Coleta",
  ENTREGA: "Entrega",
};

export function routeStopTypeLabel(type: string): string {
  return ROUTE_STOP_TYPE_LABELS[type] ?? fallbackRouteStopTypeLabel(type);
}

function fallbackRouteStopTypeLabel(type: string): string {
  const normalized = type.trim().replaceAll("_", " ").toLowerCase();
  if (!normalized) return "Parada";
  return normalized.charAt(0).toUpperCase() + normalized.slice(1);
}
